### PR TITLE
React Apollo 2.1 Migration

### PR DESCRIPTION
**Pull request by karunsingh**
- Migrating to react-apollo 2.1 from 2.0, fixing incompatibilities
- Added mutations for resolving and deleting questions on the TA side
- Added mutation for creating a new question, and retracting the user's question on the student side

### CODE REVIEW

File: client/src/components/includes/SessionQuestionsComponent.tsx [hunk 1]
```diff
@@ -24,10 +44,34 @@ class SessionQuestionsComponent extends React.Component {
         } else {
             // Disclaimer: none of us wrote this one-line magic :)
             // It is borrowed from https://stackoverflow.com/revisions/39466341/5
-            return index + ['st', 'nd', 'rd'][((index + 90) % 100 - 10) % 10 - 1] || 'th';
+            return index + ['st', 'nd', 'rd'][((index + 90) % 100 - 10) % 10 - 1] || index + 'th';
         }
     }
 
+    _onClickDelete(event: React.MouseEvent<HTMLElement>, f: Function) {
+        f({
```
Comment (Glitched) [inline client/src/components/includes/SessionQuestionsComponent.tsx]: Can we rename the Function "f" to something more descriptive? It's a little confusing at the moment

File: client/src/components/includes/AddQuestion.tsx [hunk 2]
```diff
@@ -62,7 +74,15 @@ class AddQuestion extends React.Component {
         this.setState({ redirect: true });
     }
 
-    public handleJoinClick(event: React.MouseEvent<HTMLElement>): void {
+    public handleJoinClick(event: React.MouseEvent<HTMLElement>, f: Function): void {
+        f({
```
Comment (Glitched) [inline client/src/components/includes/AddQuestion.tsx]: Here, too

File: client/src/components/includes/SessionQuestionsComponent.tsx [hunk 3]
```diff
@@ -24,10 +44,34 @@ class SessionQuestionsComponent extends React.Component {
...
+                status: 'noshow',
+                timeResolved: new Date(),
+                sessionId: this.props.sessionId,
+                answererId: userId
+            }
+        });
+    }
+
+    _onClickResolve(event: React.MouseEvent<HTMLElement>, f: Function) {
+        f({
```
Comment (Glitched) [inline client/src/components/includes/SessionQuestionsComponent.tsx]: And here

**Comment by karunsingh** [comment]
@Glitched Fixed, take a look! 👍

**Pull request merged**